### PR TITLE
add a top-level !user command with subcommands

### PR DESCRIPTION
**Pull request by ryanlerch**
This replaces the old !user command (which displayed user info) with a new top level !user command with the following sub-commands (localtime, hello, info).

Had to implement aliases for these 3 subcommands manually, as maubot wont let you alias a command-level to a command + subcommand. So these are done with a passive command.

### CODE REVIEW

File: fedora/fas.py [hunk 1]
```diff
@@ -100,19 +100,7 @@ async def group_info(self, evt: MessageEvent, groupname: str) -> None:
...
-        Returns a short line of information about the user. If no username is provided, defaults to
-        the sender of the message.
-
-        #### Arguments ####
-
-        * `username`: A Fedora Accounts username or a Matrix User ID
-           (e.g. @username:fedora.im)
-
-        """
+    async def _user_hello(self, evt: MessageEvent, username: str) -> None:
```
Comment (abompard) [inline fedora/fas.py:103]: If `username` can be `None`, let's state it in the type hints: `username: str | None`

File: fedora/fas.py [hunk 2]
```diff
@@ -128,19 +116,7 @@ async def hello(self, evt: MessageEvent, username: str) -> None:
...
-        Returns a information from Fedora Accounts about the user
-        If no username is provided, defaults to the sender of the message.
-
-        #### Arguments ####
-
-        * `username`: A Fedora Accounts username or a Matrix User ID
-           (e.g. @username:fedora.im)
-
-        """
+    async def _user_info(self, evt: MessageEvent, username: str) -> None:
```
Comment (abompard) [inline fedora/fas.py:119]: If `username` can be `None`, let's state it in the type hints: `username: str | None`

**Review by abompard — CHANGES_REQUESTED** [review]

File: fedora/fas.py [hunk 3]
```diff
@@ -158,22 +134,10 @@ async def user(self, evt: MessageEvent, username: str) -> None:
...
-
-        * `username`: A Fedora Accounts username or a Matrix User ID
-           (e.g. @username:fedora.im)
-
-        """
+    async def _user_localtime(self, evt: MessageEvent, username: str) -> None:
         await evt.mark_read()
         try:
-            user = await get_fasuser(username, evt, self.plugin.fasjsonclient)
+            user = await get_fasuser(username or evt.sender, evt, self.plugin.fasjsonclient)
```
[the remark this answers is not in the document]
Comment (ryanlerch) [reply fedora/fas.py:140]: heh -- i kinda like that option :)

moreso the other user commands default to using the sender as the default, so i figured it wouldnt hurt to do it here too.

File: tests/test_bot.py [hunk 4]
```diff
@@ -17,9 +17,7 @@ async def test_help(bot, plugin):
         "● `!pagureissue <project> <issue_id>` - return a pagure issue\n"
         "● `!whoowns <package>` - Retrieve the owner of a given package\n"
         "● `!group  <subcommand> [...]` - Query information about Fedora Accounts groups\n"
-        "● `!hello [username]` - Return brief information about a Fedora user.\n"
-        "● `!localtime <username>` - Returns the current time of the user.\n"
-        "● `!user [username]` - Return brief information about a Fedora user.\n"
+        "● `!user  <subcommand> [...]` - Get information about Fedora Accounts users\n"
```
[the remark this answers is not in the document]
Comment (ryanlerch) [reply tests/test_bot.py:20]: i was a little worried that the aliases were starting to pollute the top - level commands list. we still have !user hello as the main command.

Adding to my todolist to try to figure out a neat way to get alias help here.

**Comment by ryanlerch** [comment]
Okay, fixed:

* the type checking issues on the username params
* added the r"" string prefix on the regex

other comments i have commented back with my thoughts

**Pull request merged**